graph-builder/tests/e2e/mod.rs

```diff
@@ -6,6 +6,86 @@ use std::env;
 use test_case::test_case;
 use url::Url;
 
+pub fn sort_by_version(v: &mut Value) {
+    if !v.is_object() {
+        return;
+    }
+    let obj = v.as_object_mut().unwrap();
+
+    let mut index_map = std::collections::HashMap::<usize, usize>::new();
+
+    {
+        let mut version_index =
+            std::collections::HashMap::<String, (Option<usize>, Option<usize>)>::new();
+
+        let nodes = obj.get_mut("nodes").unwrap();
+        nodes
+            .as_array()
+            .unwrap()
+            .iter()
+            .enumerate()
+            .for_each(|(i, node)| {
+                let version = node.get("version").unwrap().as_str().unwrap();
+                version_index.insert(version.to_string(), (Some(i), None));
+            });
+
+        nodes.as_array_mut().unwrap().sort_unstable_by(|a, b| {
+            a.get("version")
+                .unwrap()
+                .as_str()
+                .cmp(&b.get("version").unwrap().as_str())
+        });
+
+        nodes
+            .as_array()
+            .unwrap()
+            .iter()
+            .enumerate()
+            .for_each(|(i, node)| {
+                let version = node.get("version").unwrap().as_str().unwrap();
+                version_index
+                    .entry(version.to_string())
+                    .and_modify(|(from, to)| {
+                        *to = Some(i);
+                        println!(
+                            "{} changed index from {} to {}",
+                            version,
+                            from.unwrap(),
+                            to.unwrap()
+                        );
+                        index_map.insert(from.unwrap(), to.unwrap());
+                    });
+            });
+    }
+
+    obj.get_mut("edges")
+        .unwrap()
+        .as_array_mut()
+        .unwrap()
+        .iter_mut()
+        .for_each(|ref mut edge: &mut serde_json::Value| {
+            if edge.as_array().unwrap().len() < 2 {
+                return;
+            }
+
+            macro_rules! rewrite_edge_index {
```

Comment:
Does it have to be defined in `for_each`? Could it be defined once instead?

Reply:
The macro makes direct use of the variable `edge` which is local to this `for_each`. I also don't foresee the usage of it outside of this location.